src/libcollections/slice.rs

```diff
@@ -1337,6 +1337,61 @@ impl<T> [T] {
         core_slice::SliceExt::sort_unstable_by_key(self, f);
     }
 
+    /// Permutes the slice in-place such that `self[mid..]` move to the
```

Comment:
Hmm, I guess I was mentally inserting "the elements in".  Considering it as "the subslice" is probably better; will update.